praca/praca.tex

```diff
@@ -245,48 +245,259 @@ \subsubsection{System wieloagentowy}
+\item \textit{MOPEnterStrategy} -- strategia decyzji o zjeździe na MOP-ie. Obecnie zaimplementowane strategie to:
+\begin{itemize}
+    \item \texttt{RANDOM\_STRATEGY}, która ma charakter testowy -- pojazdy zjeżdżają na MOP-a z prawdopodobieństwem równym $\frac{1}{2}$.
+    \item \texttt{BASIC\_STRATEGY} -- domyślna strategia dla samochodów. Losowana jest jedna liczba z przedziału 0-32400 według rozkładu jednostajnego. Pojazd wjeżdża na MOP, jeżeli od czasu poprzedniego zatrzymania minęło więcej sekund, niż wynosi wylosowana liczba.
+    \item \texttt{TRUCK\_STRATEGY} -- domyślna strategia dla pojazdów ciężarowych i autobusów. Losowana jest liczba według rozkład normalnego o średniej 14400 i odchyleniu standardowym 3600.  Pojazd wjeżdża na parking, jeżeli od poprzedniego zatrzymania minęło więcej czasu, niż wynosi wylosowana liczba.
+\end{itemize}
+\item \textit{MOPStayStrategy} -- strategia długości pobytu na MOP-ie. Obecnie zaimplementowane strategie:
+\begin{itemize}
+    \item \texttt{CAR\_STAY\_STRATEGY} -- domyślna strategia dla samochodów. Czas pobytu losowany jest według rozkładu wykładniczego o wartości oczekiwanej wynoszącej 23 minuty.
+    \item \texttt{TRUCK\_STAY\_STRATEGY} -- domyślna strategia dla pojazdów ciężarowych. Najpierw, ze względu na godzinę i czas od poprzedniego postoju, wybierany jest charakter postoju -- krótki (wymuszony przepisami, zwykle w celu tankowania, spożycia posiłku lub skorzystania z toalety) lub długi (zwykle połączony z noclegiem). Jeżeli został wybrany postój krótki, to długość pobytu jest losowana zgodnie z rozkładem normalnym o wartości oczekiwanej wynoszącej 45 minut i odchyleniu standardowym 15 minut. Dla pobytu długiego, parametry rozkładu normalnego wynoszą odpowiednio 9 godzin i półtorej godziny.
+\end{itemize}
+% TODO dopisać kolejne strategie, jeżeli powstaną
+\end{itemize}
+
+\subsubsection{Mopsim - dodanie własnych strategii}
+By przeprowadzać efektywne symulacje, zachowanie agentów powinno jak najbardziej odpowiadać rzeczywistości. Z poziomu programisty do programu Mopsim można dodać kolejne strategie. W pakiecie \texttt{strategies} znajdują się interfejsy dla każdego z rodzajów strategii -- odpowiednio \textit{TimeDistribution}, \textit{MOPEnterStrategy} i \textit{MOPStayStrategy}. By dodać strategię, należy zaimplementować dany interfejs (w każdym przypadku składający się z dwóch metod -- \texttt{getIdentifier()}, zwracającej unikalny identyfikator strategii oraz odpowiedniej funkcji decyzyjnej) oraz zmodyfikować metody klasy \texttt{StrategyUtils}, dodając do niej kolejne strategie.
 
+\section{Mopsim - opis implementacji}
+Program Mopsim został napisany z wykorzystaniem \textbf{Apache Maven}. Jest to narzędzie automatyzujące budowę oprogramowania na platformę Java. Poszczególne funkcje Mavena realizowane są poprzez wtyczki, które są automatycznie pobierane przy ich pierwszym wykorzystaniu. Plik określający sposób budowy aplikacji nosi nazwę \textbf{POM}-u.
```

Comment:
drugi raz identyczne zdanie

Reply:
to prawda ale nie wiedzialem jak zacząć

Reply:
Może "Apache Maven to narzędzie automatyzujące budowę oprogramowania na platformę Java." skoro już wcześniej piszesz że jest używany?